src/steamship/experimental/package_starters/telegram_bot.py

```diff
@@ -0,0 +1,136 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Optional, Type
+
+import requests
+from pydantic import Field
+
+from steamship import SteamshipError
+from steamship.invocable import Config, InvocableResponse, PackageService, post
+
+
+class TelegramBotConfig(Config):
+    bot_token: str = Field(description="The secret token for your Telegram bot")
+
+
+class TelegramBot(PackageService, ABC):
+
+    config: TelegramBotConfig
+
+    @classmethod
+    def config_cls(cls) -> Type[Config]:
+        """Return the Configuration class."""
+        return TelegramBotConfig
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.api_root = f"https://api.telegram.org/bot{self.config.bot_token}"
+
+    def instance_init(self):
+        """This instance init method is called automatically when an instance of this package is created. It registers the URL of the instance as the Telegram webhook for messages."""
+        webhook_url = self.context.invocable_url + "respond"
+        response = requests.get(
+            f"{self.api_root}/setWebhook",
+            params={"url": webhook_url, "allowed_updates": ["message"]},
+        )
+        if not response.ok:
+            raise SteamshipError(
+                f"Could not set webhook for bot. Webhook URL was {webhook_url}. Telegram response message: {response.text}"
+            )
+        logging.info(f"Initialized webhook with URL {webhook_url}")
+
+    @abstractmethod
+    def respond_to_text(self, message_text: str, chat_id: int, message_id: int) -> Optional[str]:
+        pass
+
+    @post("answer", public=True)
+    def answer(self, question: str, chat_session_id: Optional[str] = None) -> Dict[str, Any]:
+        """Endpoint that implements the contract for Steamship embeddable chat widgets. This is a PUBLIC endpoint since these webhooks do not pass a token."""
+        logging.info(f"/answer: {question} {chat_session_id}")
+
+        if not chat_session_id:
+            chat_session_numeric_id = 0
+        try:
+            chat_session_numeric_id = int(chat_session_id)
+        except ValueError:
+            chat_session_numeric_id = 0
+
+        message_id = 0
+
+        try:
+            response = self.respond_to_text(question, chat_session_numeric_id, message_id)
+        except SteamshipError as e:
+            response = self.response_for_exception(e)
+
+        return {
+            "answer": response,
+            "sources": [],
```

Comment:
This prevents us from delivering sources with an Ask My XYZ bot, which for that class of Agent is a feature that folks use

One option without going full experimenta.transport style is to have respond_to_text return a tuple with optional sources array as the second item

Reply:
Confused a little about the intent here as well - I thought that the "answer" wrapper for telegram bots was just to make it so that the telegram bot could also be invoked by the web ui?  

If, for now, because this is a superclass and not a transport mix-in, the point of anything that inherits from it is to deploy to Telegram, and Telegram doesn't have any special handling for a source list, then what would be the point of delivering a source array?  It would be better if you were an implementor (again, with a focus on Telegram) to pre-format the sources into the returned text, right?